crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -87,6 +87,10 @@ impl SettlementEncoder {
         &self.trades
     }
 
+    pub fn interactions(&self) -> &Vec<Arc<dyn Interaction>> {
```

Comment:
nit: Lets call this `execution_plan` since it is only interactions from the execution plan and don't include things like WETH unwraps.